tests/activity_api_reset_test.go

```diff
@@ -0,0 +1,328 @@
+	return func(ctx workflow.Context) (string, error) {
+
+		var ret string
+		err := workflow.ExecuteActivity(workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+			ActivityID:             "activity-id",
+			DisableEagerExecution:  true,
+			StartToCloseTimeout:    s.startToCloseTimeout,
+			ScheduleToCloseTimeout: s.scheduleToCloseTimeout,
+			RetryPolicy:            s.activityRetryPolicy,
+		}), activityFunction).Get(ctx, &ret)
+		return "done!", err
+	}
+}
+
+func (s *ActivityApiResetClientTestSuite) TestActivityResetApi_Acceptance() {
+	// activity reset is called after multiple attempts,
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	var activityReseted atomic.Bool
+	activityCompleteSignal := make(chan struct{})
+	var activityCompleted atomic.Int32
+
+	activityFunction := func() (string, error) {
+		if activityReseted.Load() == false {
+			activityErr := errors.New("bad-luck-please-retry")
+			activityCompleted.Add(1)
+			return "", activityErr
+		}
+
+		activityCompleted.Add(1)
+		s.WaitForChannel(ctx, activityCompleteSignal)
+		return "done!", nil
+	}
+
+	workflowFn := s.makeWorkflowFunc(activityFunction)
+
+	s.Worker().RegisterWorkflow(workflowFn)
+	s.Worker().RegisterActivity(activityFunction)
+
+	wfId := testcore.RandomizeStr("wfid-" + s.T().Name())
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:        wfId,
+		TaskQueue: s.TaskQueue(),
+	}
+
+	workflowRun, err := s.SdkClient().ExecuteWorkflow(ctx, workflowOptions, workflowFn)
+	s.NoError(err)
+
+	// wait for activity to start/fail few times
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		assert.NoError(t, err)
+		assert.Equal(t, 1, len(description.PendingActivities))
+		assert.True(t, activityCompleted.Load() > 1)
+	}, 5*time.Second, 200*time.Millisecond)
+
+	resetRequest := &workflowservicepb.ResetActivityByIdRequest{
+		Namespace:  s.Namespace(),
+		WorkflowId: workflowRun.GetID(),
+		ActivityId: "activity-id",
+		NoWait:     true,
+	}
+	resp, err := s.FrontendClient().ResetActivityById(ctx, resetRequest)
+	s.NoError(err)
+	s.NotNil(resp)
+
+	activityReseted.Store(true)
+
+	// wait for activity to be running
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		assert.NoError(t, err)
+		assert.Equal(t, 1, len(description.PendingActivities))
+		assert.Equal(t, enumspb.PENDING_ACTIVITY_STATE_STARTED, description.PendingActivities[0].State)
+		// also verify that the number of attempts was reset
+		assert.Equal(t, int32(1), description.PendingActivities[0].Attempt)
+	}, 5*time.Second, 100*time.Millisecond)
+
+	// let activity finish
+	activityCompleteSignal <- struct{}{}
+
+	// wait for workflow to complete
+	var out string
+	err = workflowRun.Get(ctx, &out)
+
```

Comment:
nit: no need for newline (other error checks are right below the error)

_(same below)_

Reply:
removed